src/doc/src/reference/build-scripts.md

```diff
@@ -282,6 +282,16 @@ It is recommended to group the `cargo::rustc-check-cfg` and
 [`cargo::rustc-cfg`][option-cfg] instructions as closely as possible in order to
 avoid typos, missing check-cfg, stalled cfgs...
 
+It is also possible to add to the list of expected config names and values using the special
+[`check-cfg`][lint-table-check-cfg] sub-config under the `[lints.rust.unexpected_cfgs]` lint:
+
+```toml
+[lints.rust]
+unexpected_cfgs = { level = "warn", check-cfg = ["cfg(foo, values(\"bar\"))"] }
+```
+
+[lint-table-check-cfg]: manifest.md#special-check-cfg
```

Comment:
With this change, the way I see it working is
- `build.rs` is used if using `cargo::rustc-cfg`
- `Cargo.toml` is used for everything else

In that case, this doesn't really fit in the build script documentation.

Reply:
Ideally yes, it shouldn't be in the build script documentation but we currently don't have a canonical place for check-cfg (`rustc` link directly to this doc) and more importantly I see this as an alternative to the instruction, so mentioning it doesn't seems that out of place.

Reply:
> Ideally yes, it shouldn't be in the build script documentation but we currently don't have a canonical place for check-cfg (rustc link directly to this doc) 

This is a rust lint config and belongs in their documentation.  

> I see this as an alternative to the instruction, so mentioning it doesn't seems that out of place.

I gave reasons for why I disagree.  Could you address those?

Reply:
I've removed this documentation update, the build-script doc should only be about build-script.